src/Courses/TopicGrades/ExportAllButton.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { Button, ButtonGroup, CircularProgress, ClickAwayListener, Grow, MenuItem, MenuList, Paper, Popper } from '@material-ui/core';
+import { startExportOfTopic } from '../../APIInterfaces/BackendAPI/Requests/CourseRequests';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import _ from 'lodash';
+import { useRouteMatch } from 'react-router-dom';
+
+interface ExportAllButtonProps {
+    topicId: number;
+    userId: number | undefined;
+}
+
+enum LoadingState {
+    UNTOUCHED,
+    LOADING,
+    SUCCESS,
+    ERROR
+}
+
+enum ButtonOptions {
+    EXPORT_ALL = 'Export All (.zip)',
+    // TODO: Implement
+    // EXPORT_ALL_NO_SOLUTIONS = 'Export All without Solutions (.zip)',
+    RECALCULATE = 'Recreate Archive',
```

Comment:
When the above is implemented you'll need two recreate options

The options probably make more sense to be:
Download student grades archive
Recreate student grades archive with solutions
Recreate student grades archive without solutions

Reply:
- Create PDF bundle with solutions
- Create PDF bundle without solutions
- Download PDF bundle (should this display whether or not the bundle has solutions?)